include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2178,6 +2178,13 @@ class binary_reader
             std::vector<char_int_type> bjdx = {'[', '{', 'S', 'H', 'T', 'F', 'N', 'Z'}; // excluded markers in bjdata optimized type
 
             result.second = get();  // must not ignore 'N', because 'N' maybe the type
+            if (JSON_HEDLEY_UNLIKELY( input_format == input_format_t::bjdata && std::find(bjdx.begin(), bjdx.end(), result.second) != bjdx.end() ))
+            {
+                auto last_token = get_token_string();
+                return sax->parse_error(chars_read, last_token, parse_error::create(112, chars_read,
+                                        exception_message(input_format, concat("marker 0x", last_token, " is not a permitted optimized array type"), "type"), nullptr));
```

Comment:
Is 112 the correct exception ID and should the [documentation](https://json.nlohmann.me/home/exceptions/#jsonexceptionparse_error112) be updated?

Reply:
Then I suggest you add BJData to the documentation for 112 as part of #3464.

"Not all types of CBOR, MessagePack, or BJData are supported. This exception occurs if an unsupported byte was read."

Reply:
👍

Reply:
Done in 09d4c3b4a598b73e96b98fcb4a699466cda3cd5a.